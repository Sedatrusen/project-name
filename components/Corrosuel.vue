<template>
  <div>
    <b-carousel
      v-model="slide"
      class="carousel-1"
      :interval="4000"
      controls
      indicators
      background="#ababab"
      img-width="1024"
      img-height="480"
      style="text-shadow: 1px 1px 2px #333"
    >
      <!-- Text slides with image -->
      <b-carousel-slide
        v-for="item in items"
        :key="item.id"
        :img-src="item.image"
        ><div class="info">
          <img class="resimler" :src="item.image2" alt="" />
          <p>{{ item.parca }}</p>
          <button class="button">{{ item.buton }}</button>
        </div>
      </b-carousel-slide>
    </b-carousel>
  </div>
</template>

<script>
import { mapState } from 'vuex'
export default {
  data() {
    return {
      slide: 0,
      sliding: null,
    }
  },
  computed: {
    ...mapState({ items: (state) => state.carouselitems.carouselitems }),
  },
  created() {
    this.$store.dispatch('carouselitems/addcarouselitems')
  },
}
</script>
<style scoped>
.resimler {
  height: 150px;
  width: 250px;
}
.carousel-1 {
  display: flex;
  justify-items: start;
  flex: 1;
  align-items: flex-start;
}

.button {
  height: 40px;
  width: 280px;
  background-color: #0e86ca;
  border: none;
  color: white;
  font-family: 'Open Sans', Helvetica, Arial, sans-serif;
}
.button:hover {
  background-color: #83cefa;
}
.info {
  margin-right: 50rem;
}
</style>
