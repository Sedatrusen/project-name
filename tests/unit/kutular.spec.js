import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import Corrosuel from '@/components/Kutular.vue'

const localVue = createLocalVue()

localVue.use(Vuex)

describe('Kutular component', () => {
  let state
  let store
  let actions

  beforeEach(() => {
    state = {
      kutular: jest.fn(),
    }
    actions = {
      addkutular: jest.fn(),
    }
    store = new Vuex.Store({
      state,
      actions,
    })
  })

  it('korosel itemleri set etme', () => {
    const wrapper = shallowMount(Corrosuel, {
      localVue,
      store,
    })
    expect(wrapper.exists()).toBe(true)
    expect(wrapper).toMatchSnapshot()
  })
})
