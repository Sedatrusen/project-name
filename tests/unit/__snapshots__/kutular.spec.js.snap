// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Kutular component korosel itemleri set etme 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
